tests/testthat/test-plot_with_settings.R

```diff
@@ -58,3 +58,468 @@ testthat::test_that("if teal.plot_dpi is not an integer value then get_plot_dpi
+    ),
+    NA
+  )
+})
+
+testthat::test_that("clean_brushedPoints removal of NA points", {
+  testthat::expect_identical(nrow(data), 20L)
+  testthat::expect_identical(nrow(clean_brushedPoints(data, brush)), 11L)
+  data$AGE[1:10] <- NA
+  testthat::expect_identical(nrow(clean_brushedPoints(data, brush)), 6L)
+})
+
+download_srv_args <- list(
+  id = "STH",
+  plot_reactive = function() plot(1),
+  plot_type = function() "base",
+  plot_w = function() 250,
+  default_w = function() 300,
+  plot_h = function() 200,
+  default_h = function() 300
+)
+
+testthat::test_that("type_download_srv download all types of files with default name", {
+  shiny::testServer(
+    teal.widgets:::type_download_srv,
+    args = download_srv_args,
+    expr = {
+      for (type in c("png", "pdf", "svg")) {
+        session$setInputs(`file_format` = type)
+        session$setInputs(`data_download` = 1)
+        testthat::expect_true(file.exists(output$data_download))
+        testthat::expect_identical(
+          basename(output$data_download),
+          paste0(".", type)
+        )
+      }
+    }
+  )
+})
+
+testthat::test_that("type_download_srv download all types of files and change the name", {
+  shiny::testServer(
+    teal.widgets:::type_download_srv,
+    args = download_srv_args,
+    expr = {
+      for (type in c("png", "pdf", "svg")) {
+        session$setInputs(`file_format` = type)
+        session$setInputs(`data_download` = 1)
+        session$setInputs(`file_name` = "RANDOM_NAME")
+        testthat::expect_true(file.exists(output$data_download))
+        testthat::expect_identical(
+          basename(output$data_download),
+          paste0("RANDOM_NAME.", type)
+        )
+      }
+    }
+  )
+})
+
+testthat::test_that("type_download_srv download a png file with different dimensions", {
+  shiny::testServer(
+    teal.widgets:::type_download_srv,
+    args = download_srv_args,
+    expr = {
+      session$setInputs(`file_format` = "png")
+      session$setInputs(`data_download` = 1)
+      testthat::expect_identical(
+        attr(png::readPNG(output$data_download, info = TRUE), "info")$dim,
+        c(250L, 200L)
+      )
+    }
+  )
+})
+
+testthat::test_that("type_download_srv download a png file with NULL input dimensions", {
+  shiny::testServer(
+    teal.widgets:::type_download_srv,
+    args = list(
+      id = "STH",
+      plot_reactive = function() plot(1),
+      plot_type = function() "base",
+      plot_w = function() NULL,
+      default_w = function() 350,
+      plot_h = function() NULL,
+      default_h = function() 300
+    ),
+    expr = {
+      session$setInputs(`file_format` = "png")
+      session$setInputs(`data_download` = 1)
+      testthat::expect_identical(
+        attr(png::readPNG(output$data_download, info = TRUE), "info")$dim,
+        c(350L, 300L)
+      )
+    }
+  )
+})
+
+
+testthat::test_that("plot_with_settings_srv assert error", {
```

Comment:
All those assertions: maybe do it using list and a loop, like in example before (print_plot for example)?

Reply:
No, please look at it more deeply.
The tested function is a moduleServer that is why we call it separately.

Reply:
I meant something like that:
```
argsss <- list(
  list(id = "STH", plot_r = function() plot(1), height = "20"),
  list(id = "STH", plot_r = function() plot(1), width = c(20, 100, 200)),
  list(id = "STH", plot_r = NULL),
  list(id = "STH", plot_r = function() plot(1), dblclicking = 20)
)

for (argss in argsss) {
  testthat::expect_error(
    shiny::testServer(
      teal.widgets:::plot_with_settings_srv,
      args = argss
    ),
    "Assertion"
  )
}
```

Reply:
Now I understand, agree It will be better:)